build/build_suite_test.go

```diff
@@ -179,4 +181,170 @@ var _ = Describe("Test Build utilities:", func() {
+				Name:      "foo",
+				Namespace: "bar",
+			},
+			Type: v1.SecretTypeTLS,
+			Data: map[string][]byte{
+				v1.TLSCertKey: fakeInput,
+			},
+		}
+		d := &Deployment{
+			Secrets: dSecrets,
+		}
+		name := manager.SystemEndpointSecretName
+		username := os.Getenv(manager.UsernameKey)
+		password := os.Getenv(manager.PasswordKey)
+		secret2 := v1.Secret{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Secret",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: db.namespace,
+			},
+			Type: v1.SecretTypeOpaque,
+			Data: map[string][]byte{
+				manager.UsernameKey: []byte(username),
+				manager.PasswordKey: []byte(password),
+			},
+			StringData: map[string]string{
+				manager.RegionNameKey:  os.Getenv(manager.RegionNameKey),
+				manager.DomainNameKey:  os.Getenv(manager.DomainNameKey),
+				manager.ProjectNameKey: os.Getenv(manager.ProjectNameKey),
+				manager.AuthUrlKey:     os.Getenv(manager.AuthUrlKey),
+				manager.InterfaceKey:   os.Getenv(manager.InterfaceKey),
+			},
+		}
+		keystoneRegion := os.Getenv(manager.KeystoneRegionNameKey)
+		if keystoneRegion != "" {
+			secret2.StringData[manager.KeystoneRegionNameKey] = keystoneRegion
+		}
+		want := append(dSecrets, &secret2)
+		err := db.buildEndpointSecret(d)
+		Expect(err).To(BeNil())
+		Expect(d.Secrets).To(Equal(want))
+	})
+
+	Describe("Test isInterfaceInUse", func() {
+		It("when bonds Members has the ifname interface", func() {
+			ifname := "ifname"
+			info := &starlingxv1.InterfaceInfo{}
+			bonds := make(starlingxv1.BondList, 1)
+			bonds[0] = starlingxv1.BondInfo{
+				Members: []string{"1", "ifname", "2"},
+			}
+			info.Bond = bonds
+			out := isInterfaceInUse(ifname, info)
+			Expect(out).To(Equal(true))
+		})
+	})
+
+	Describe("Test isInterfaceInUse", func() {
+		It("when VLAN lower defines the ifname interface", func() {
+			ifname := "ifname"
+			info := &starlingxv1.InterfaceInfo{}
+			info.VF = make(starlingxv1.VFList, 1)
+			bonds := make(starlingxv1.BondList, 1)
+			info.VLAN = make(starlingxv1.VLANList, 1)
+			info.VLAN[0].Lower = "ifname"
+			info.VF[0].Lower = "VFLower"
+			bonds[0] = starlingxv1.BondInfo{
+				Members: []string{"1", "2"},
+			}
+			info.Bond = bonds
+			out := isInterfaceInUse(ifname, info)
+			Expect(out).To(Equal(true))
+		})
+	})
+
+	Describe("Test isInterfaceInUse", func() {
+		It("when VF lower defines the ifname interface", func() {
+			ifname := "ifname"
+			info := &starlingxv1.InterfaceInfo{}
+			info.VF = make(starlingxv1.VFList, 1)
+			bonds := make(starlingxv1.BondList, 1)
+			info.VLAN = make(starlingxv1.VLANList, 1)
+			info.VLAN[0].Lower = "VLANLower"
+			info.VF[0].Lower = "ifname"
+			bonds[0] = starlingxv1.BondInfo{
+				Members: []string{"1", "2"},
+			}
+			info.Bond = bonds
+			out := isInterfaceInUse(ifname, info)
+			Expect(out).To(Equal(true))
+		})
+	})
+
+	Describe("Test isInterfaceInUse", func() {
+		It("when interface is not in use", func() {
```

Comment:
missing "Context" intentionally?
i think its good to have from readability perspective.